src/TypeLevel/RowList/RowList.purs

```diff
@@ -0,0 +1,21 @@
+module TypeLevel.RowList.Unordered where
+
+import TypeLevel.Nat
+import Type.RowList
+import Prim.TypeError (class Fail, Text)
+import Data.Symbol
+
+-- | Uniqueness constraint on the labels of an unordered RowList
+class AllUniqueLabels :: forall (k :: Type). RowList k -> Constraint
+class AllUniqueLabels list
+
+instance
+  Fail (Text "Labels are not unique!") =>
+  AllUniqueLabels (Cons k a (Cons k a'  xs))
```

Comment:
Wait why does the clause assume that? It's possible that I'm forgetting something (been away from this for a few days) but I don't see how the clause supposes that the RowLists are ordered at all? 

The instance definitions should express the proposition:

A RowList has unique labels <-> [(it is empty V it contains one element V (the head cons'd to the tail of the tail has unique labels & the tail has unique labels)) & the labels of the first two elements are distinct]

I don't think the order of the RowList matters (unless I ordered the definitions incorrectly so that it somehow does)? Consider (using normal list syntax for simplicity):

`z : d  : a : d : []`

To solve the `AllUniqueLabels` constraint, the compiler has to solve: 

1) `AllUniqueLabels (z : a : d : [])` (not expanding this b/c it obviously can be solved)
2) `AllUniqueLabels (d : a : d [])`
2a) `AllUniqueLabels (d : d : [])` - Fails here 

That should work for every RowList no matter the order of the labels, shouldn't it?

Reply:
Thanks for the elaboratino, I figure it out at [b60cb1a](https://github.com/gnumonik/cardano-transaction-lib/pull/1/commits/b60cb1acc454bf0e89574deb9f84666a5a7fb98d) :D forgot to remove the comment :(